x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/date/DateExtract.java

```diff
@@ -109,16 +108,7 @@ protected TypeResolution resolveType() {
         if (childrenResolved() == false) {
             return new TypeResolution("Unresolved children");
         }
-        TypeResolution resolution = argumentTypesAreSwapped(
-            children().get(0).dataType(),
-            children().get(1).dataType(),
-            DataTypes::isString,
-            sourceText()
-        );
-        if (resolution.unresolved()) {
-            return resolution;
-        }
-        resolution = isStringAndExact(children().get(0), sourceText(), TypeResolutions.ParamOrdinal.FIRST);
+        TypeResolution resolution = isStringAndExact(children().get(0), sourceText(), TypeResolutions.ParamOrdinal.FIRST);
```

Comment:
super-duper-nit (and kinda out of scope): you could use the cool and new `and` combinator:
```suggestion
        return isStringAndExact(children().get(0), sourceText(), TypeResolutions.ParamOrdinal.FIRST)
            .and(isDate(children().get(1), sourceText(), TypeResolutions.ParamOrdinal.SECOND));
```

Reply:
Applied here and in `DateTrunc` too, thanks.